blockchain/fastsync.go

```diff
@@ -212,34 +213,32 @@ func worker(cache *utxoCache, jobs <-chan []byte, results chan<- *result) {
 // downloadUtxoSet will attempt to connect to make an HTTP GET request to the
 // provided sources one at a time and download the UTXO set to the provided path.
 // If a proxy is provided it will use it for the HTTP connection.
-func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) error {
-	dialFunc := net.Dial
-	if proxy != nil {
-		dialFunc = proxy.Dial
-	}
-	tr := &http.Transport{
-		Dial: dialFunc,
-	}
-	client := &http.Client{Transport: tr}
+func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) (string, error) {
+	var fileName string
+
 	for _, src := range sources {
-		resp, err := client.Get(src)
-		if err != nil {
-			continue
-		}
-		if resp.StatusCode != http.StatusOK {
-			continue
-		}
-		file, err := os.Create(pth)
-		if err != nil {
-			return err
-		}
 		log.Infof("Downloading UTXO set from %s", src)
-		_, err = io.Copy(file, resp.Body)
-		if err != nil {
-			return err
+		retry.Do(
+			func() error {
+				resp, err := grab.Get(pth, src)
```

Comment:
Is there a reason to depend on the environment proxy here (`http.ProxyFromEnvironment`) instead of using the provided proxy? I looked at it but didn't immediately see how to configure a socks proxy with Get.

Reply:
I can look into adding Proxy support for grab. For now I think this is a reasonable tradeoff till I add it to the library. Most people aren't using proxies for their bchd instance.

Reply:
I think that's reasonable, but maybe mark it a comment so it's documented and won't be overlooked in the future.

Reply:
Added comment about missing proxy support in grab.